qiskit/circuit/quantumcircuit.py

```diff
@@ -4129,7 +4158,7 @@ def _unsorted_parameters(self) -> set[Parameter]:
         """
         # This should be free, by accessing the actual backing data structure of the table, but that
         # means that we need to copy it if adding keys from the global phase.
-        return self._parameter_table.get_keys()
+        return set(self._data.get_params_unsorted())
```

Comment:
I think we need to take some care here - at an absolute minimum, the comment here is obsoleted by this, and the docstring of the function becomes wrong as well.  I don't know how much this enters into performance now too, since it's creating an extra collection.

Can we not at least return a `set` directly from `get_params_unsorted`?

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12459/commits/252a089e9549fefa707dc85a22b91bce655ad188